tests/Integrations/WordPress/V5_5/CommonScenariosTest.php

```diff
@@ -0,0 +1,395 @@
+                                            ]),
+                                        SpanAssertion::exists('wp_head')
+                                            ->withChildren([
+                                                SpanAssertion::exists('wp_print_head_scripts'),
+                                                SpanAssertion::exists('wpdb.query')
+                                                    ->withChildren([
+                                                        SpanAssertion::exists('mysqli_query'),
+                                                    ]),
+                                                SpanAssertion::exists('wpdb.query')
+                                                    ->withChildren([
+                                                        SpanAssertion::exists('mysqli_query'),
+                                                    ]),
+                                                SpanAssertion::exists('wpdb.query')
+                                                    ->withChildren([
+                                                        SpanAssertion::exists('mysqli_query'),
+                                                    ]),
+                                            ]),
+                                    ]),
+                            ]),
+                        SpanAssertion::exists('wp_maybe_load_embeds'),
+                        SpanAssertion::exists('wp_maybe_load_widgets'),
+                        SpanAssertion::exists('wpdb.query')
+                            ->withChildren([
+                                SpanAssertion::exists('mysqli_query'),
+                            ]),
+                        SpanAssertion::exists('wpdb.query')
+                            ->withChildren([
+                                SpanAssertion::exists('mysqli_query'),
+                            ]),
+                        SpanAssertion::exists('wpdb.query')
+                            ->withChildren([
+                                SpanAssertion::exists('mysqli_query'),
+                            ]),
+                        SpanAssertion::exists('wpdb.query')
+                            ->withChildren([
+                                SpanAssertion::exists('mysqli_query'),
+                            ]),
+                    ]),
+                ],
+                'A GET request with an exception' => [
+                    SpanAssertion::build(
+                        'wordpress.request',
+                        'wordpress_55_test_app',
+                        'web',
+                        'GET /error'
+                    )->withExactTags([
+                        'http.method' => 'GET',
+                        'http.url' => 'http://localhost:9999/error',
+                        // WordPress doesn't appear to automatically set the proper error code
+                        'http.status_code' => '200',
```

Comment:
This one is not a blocker, but something to think about for a separate PR. We should probably report 500 for uncaught exceptions even if WP doesn't. Otherwise they will look like normal traces in the UI (especially since the exception isn't attached to the root span yet). 

IIRC we discussed this idea about another framework, but I can't remember which one. We might want to come up with a more generic solution that will work across all frameworks, so I'll start an internal discussion so we can iron out some details. :)

Reply:
Good idea to start this conversation, I am unsure I fully agree.

> We should probably report 500 for uncaught exceptions even if WP doesn't.

I think we should report what the user sees. If the app returns 200, then we should show 200. Otherwise we give a misleading info. E.g. a developer running `curl <url> -v` would see a `500` trace in the UI and a `200` response dumped by curl. Still, they should look different...read on :)

> Otherwise they will look like normal traces in the UI (especially since the exception isn't attached to the root span yet).

This is a very good point, though. Even if we return `200`, we can still (and should) mark the trace as an error. This, though, should not be the framework's integration responsibility, and it will be provided by the work that we are doing for generic error reporting improvements.

WDYT?

Reply:
Good point - sounds good to me. 👍